recipes/scc/all/conanfile.py

```diff
@@ -0,0 +1,89 @@
+from conan import ConanFile
+from conan.tools.scm import Version
+from conan.tools.files import get
+from conans import tools, CMake
+from conan.errors import ConanInvalidConfiguration
+import functools
+
+required_conan_version = ">=1.43.0"
+
+class SystemcComponentsConan(ConanFile):
+    name = "scc"
+    description = """A light weight productivity library for SystemC and TLM 2.0"""
+    homepage = "https://minres.github.io/SystemC-Components"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    topics = ("systemc", "modeling", "tlm", "scc")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "fPIC": [True, False],
+        "sc_with_phases_callbacks": [True, False],
+        "sc_with_phases_callback_tracing": [True, False]
+    }
+    default_options = {
+        "fPIC": True,
+        "sc_with_phases_callbacks": False,
+        "sc_with_phases_callback_tracing": False
+    }
+    generators = "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    # no exports_sources attribute, but export_sources(self) method instead
+    # this allows finer grain exportation of patches per version
+    def export_sources(self):
+        self.copy("CMakeLists.txt")
+            
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 11)
+        if self.settings.os == "Macos":
+            raise ConanInvalidConfiguration(f"{self.name} is not suppported on {self.settings.os}.")
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "7":
+            raise ConanInvalidConfiguration("GCC < version 7 is not supported")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version],
+                  destination=self._source_subfolder, strip_root=True)
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.16.2")
```

Comment:
Seems out of place, this is the version CCI is running AFAIK... 83862d79897135cda8bc6c97f1aa67e8395c502a does not give much insight why

https://github.com/conan-io/conan-center-index/blob/master/docs/faqs.md#why-recipes-that-use-build-tools-like-cmake-that-have-packages-in-conan-center-do-not-use-it-as-a-build-require-by-default

Reply:
Conan Center Index runs CMake 3.15 as minimum version, however, systemC requires 3.16 at least.

Reply:
Ahh, I thought it was higher! Looking at the docs more closely... it some linux platform https://github.com/conan-io/conan-center-index/blob/master/docs/supported_platforms_and_configurations.md#linux 👍 that make sense

Reply:
@staskau I suggest you using `tool_requires("3.24.0")` instead. https://docs.conan.io/en/latest/devtools/build_requires.html#tool-requirements

Reply:
Thanks for the tip. I'll switch to tool_requires in the next upload (I'm currently waiting for an update to the system-components library that we'd like to get in before the roll-out in conan-center).
Sorry, but I didn't understand why we should restrict to version 3.24.0 when the older cmake version 3.16 is also sufficient for the successful build?

Reply:
just to avoid conflict use the latest version, which is usually recommended to use